maps/src/maputils.cxx

```diff
@@ -19,19 +19,38 @@
 
 using namespace G3Units;
 
-G3SkyMapPtr GetMaskMap(G3SkyMapConstPtr m)
+G3SkyMapPtr GetMaskMap(G3SkyMapConstPtr m, bool ignore_nans, bool ignore_infs)
 {
 	G3SkyMapPtr mask = m->Clone(false);
 
 	for (size_t i = 0; i < m->size(); i++) {
-		if (m->at(i) == 0)
+		double v = m->at(i);
+		if (ignore_nans && v != v)
+			continue;
+		if (ignore_infs && !std::isfinite(v))
+			continue;
+		if (v == 0)
 			continue;
 		(*mask)[i] = 1.0;
 	}
 
 	return mask;
 }
 
+G3SkyMapPtr ApplyMask(G3SkyMapPtr m, G3SkyMapConstPtr mask, bool inverse)
```

Comment:
This should either be a public API (prototype in the header) or declared static.